src/virtualenv/util/path/_sync.py

```diff
@@ -75,6 +78,41 @@ def onerror(func, path, exc_info):
     shutil.rmtree(ensure_text(str(dest)), ignore_errors=True, onerror=onerror)
 
 
+@contextmanager
+def atomic_open_for_write(target, binary=False, newline=None, encoding=None):
```

Comment:
Why not just write_atomic? Why separate open and write?

Reply:
You're implementing functionality we're not using, I'd rather only do as much as needed...